tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -1388,6 +1388,151 @@ int main(int argc, char **argv)
                                    S2N_ERR_BAD_MESSAGE);
      }
 
+     /**
+     *= https://tools.ietf.org/rfc/rfc8446#4.2.8
+     *= type=test
+     *# Upon receipt of this extension in a HelloRetryRequest, the client
+     *# MUST verify that (1) the selected_group field corresponds to a group
+     *# which was provided in the "supported_groups" extension in the
+     *# original ClientHello
+     **/
```

Comment:
Is this the right place / way to test this? Shouldn't we test it when we test the extension, in s2n_client_key_share_extension_test? It looks like most of those tests don't do the full handshake.

Reply:
Since both checks are now in `s2n_server_hello_retry.c`, I left the test where it was.